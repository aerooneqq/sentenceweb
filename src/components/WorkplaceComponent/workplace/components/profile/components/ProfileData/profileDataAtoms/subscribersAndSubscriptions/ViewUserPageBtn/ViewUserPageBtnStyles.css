.viewUserPageBtn{ 
    width: 80px;
    height: 30px;

    background: gray;

    border-style: solid;
    border-radius: 7px;
    border-color: gray;
    border-width: 1px;
    
    margin-left: 10px;

    font-family: Arial;
    color: white;
}

.viewUserPageBtn:focus{ 
    outline: none;
}

.viewUserPageBtn:hover{ 
    background: linear-gradient(-120deg,#707070, #808080)
}