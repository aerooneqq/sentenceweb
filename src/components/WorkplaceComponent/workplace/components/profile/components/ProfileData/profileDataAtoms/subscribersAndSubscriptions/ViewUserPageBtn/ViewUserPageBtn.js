import React, {Component} from "react";

//Styles
import "./ViewUserPageBtnStyles.css"

/**
 * PROPS LIST: 
 * 1) viewUserPage
 * 2) userID
 */
export default class ViewUserPageBtn extends Component { 

    constructor(props) { 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick() { 
        this.props.viewUserPage(this.props.userID);
    }

    render() { 
        return( 
            <button className = "viewUserPageBtn"
                    onClick = {this.handleBtnClick}>
                View
            </button>
        )
    }
}