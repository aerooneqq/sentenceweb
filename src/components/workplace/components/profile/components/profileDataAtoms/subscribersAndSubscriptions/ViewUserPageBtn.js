import React, {Component} from "react";

//Styles
import "./styles/ViewUserPageBtnStyles.css"

export default class ViewUserPageBtn extends Component{ 
    constructor(props){ 
        super(props);
    }

    render(){ 
        return( 
            <button className = "viewUserPageBtn">
                View
            </button>
        )
    }
}