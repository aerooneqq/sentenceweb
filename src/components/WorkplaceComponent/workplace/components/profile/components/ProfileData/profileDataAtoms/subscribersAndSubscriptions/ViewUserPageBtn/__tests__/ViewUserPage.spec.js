import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import ViewUserPageBtn from "../ViewUserPageBtn";

const propsSpy = jest.fn(() => {});
const props = { 
    userID: 1,
    viewUserPage: propsSpy
}

describe("ViewUserPageBtn", () => { 
    it("should render correctly", () => {
        expect(shallowToJson(shallow(<ViewUserPageBtn {...props}/>))).toMatchSnapshot();
    });

    it("should call viewUserPage from props when handleBtnClick is clicked", () => { 
        const viewUserPage = shallow(<ViewUserPageBtn {...props}/>);

        viewUserPage.instance().handleBtnClick();

        expect(propsSpy).toBeCalled();
    });

    it("should call handleBtnClick when clicked", () => { 
        const spy = jest.spyOn(ViewUserPageBtn.prototype, "handleBtnClick");
        const viewUserPage = shallow(<ViewUserPageBtn {...props}/>); 

        viewUserPage.find("button").simulate("click");

        expect(spy).toBeCalled();
    });
});