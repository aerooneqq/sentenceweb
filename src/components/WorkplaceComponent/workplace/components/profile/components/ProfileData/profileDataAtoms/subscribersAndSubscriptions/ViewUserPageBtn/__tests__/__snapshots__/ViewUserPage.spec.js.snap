// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewUserPageBtn should render correctly 1`] = `
<button
  className="viewUserPageBtn"
  onClick={[Function]}
>
  View
</button>
`;
